Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift

```diff
@@ -351,6 +332,59 @@ open class _ChatMessageContentView<ExtraData: ExtraDataTypes>: _View, ThemeProvi
             ]
         }
 
+        // Main container
+
+        mainContainer.alignment = .bottom
+        mainContainer.isLayoutMarginsRelativeArrangement = true
+        mainContainer.layoutMargins.top = 0
+        insertSubview(mainContainer, at: 0)
+
+        let mainContainerSubviews = [
+            authorAvatarView ?? authorAvatarSpacer,
+            errorIndicatorContainer,
+            bubbleThreadMetaContainer
+        ].compactMap { $0 }
+
+        if options.contains(.flipped) {
+            mainContainer.addArrangedSubviews(mainContainerSubviews.reversed())
+
+            if let errorIndicator = errorIndicatorView {
+                mainContainer.setCustomSpacing(
+                    .init(-errorIndicator.intrinsicContentSize.width / 2),
```

Comment:
What's the purpose of this? Do you want the view to be centered somehow? You can just add an additional constraint for that, no need to do this kind magic of magic imo.

Reply:
The error indicator is embedded into `mainContainer` where all the built-in constraints are `.required`. So setting the custom spacing is the easiest way.

The obvious solution with error indicator at root level with `centerX = mainContainer.leading/trailing` anchor has issues:
- for messages from current user (aligned to the right and without avatar) the error indicator is partially outside the screen
- when user avatar is shown we want error indicator to be in-between the author avatar and message content I guess ([content - error - avatar]) so the messages are aligned. Having error indicator inside `mainContainer` makes it possible + also handles the case when the cell is flipped

Reply:
All right, thank you for explaining this to me 👍